app/code/Magento/CheckoutAgreements/view/frontend/web/js/view/checkout-agreements.js

```diff
@@ -26,6 +26,17 @@ define(
             modalWindow: null,
 
             /**
+             * build a unique id for the term checkbox
+             *
+             * @param parents
+             * @param agreementId
+             */
+            getCheckboxId: function(parents, agreementId) {
+              var paymentMethodName = (parents[2].item) ? parents[2].item.method : '';
```

Comment:
Thanks for feedback.

> $parents[2] is the view model from the great-grandparent context
http://knockoutjs.com/documentation/binding-context.html

I don't like this myself, but its the only way I figured out to access the current payment method name. I'll see what I can do, to improve it.